File: 151002/Kragel/distributed/publisher/src/main/java/by/bsuir/publisher/service/mapper/CreatorMapper.java
package by.bsuir.publisher.service.mapper;

import by.bsuir.publisher.dto.request.CreatorRequestDto;
import by.bsuir.publisher.dto.response.CreatorResponseDto;
import by.bsuir.publisher.model.Creator;
import org.mapstruct.*;

import java.util.Collection;
import java.util.List;

@Mapper(componentModel = "spring")
public interface CreatorMapper {
    @Mapping(target = "firstname", source = "firstName")
    @Mapping(target = "lastname", source = "lastName")
    CreatorResponseDto toDto(Creator entity);

    @Mapping(target = "id", ignore = true)
    @Mapping(target = "firstName", source = "firstname")
    @Mapping(target = "lastName", source = "lastname")
    Creator toEntity(CreatorRequestDto creatorRequestDto);

    List<CreatorResponseDto> toDto(Collection<Creator> entities);

    List<Creator> toEntity(Collection<CreatorRequestDto> entities);

    @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
    @Mapping(target = "id", ignore = true)
    @Mapping(target = "firstName", source = "firstname")
    @Mapping(target = "lastName", source = "lastname")
    Creator partialUpdate(CreatorRequestDto creatorRequestDto, @MappingTarget Creator creator);
}

File: 151003/Alekseeva/TaskKafka/Module.Publisher/Mappers/CreatorMapper.cs
using Publisher.Dto.Request.CreateTo;
using Publisher.Dto.Response;
using Publisher.Models;
using Riok.Mapperly.Abstractions;
namespace Publisher.Mappers;

[Mapper]
public static partial class CreatorMapper
{
    public static partial Creator ToEntity(this CreateCreatorRequestTo createCreatorRequestTo);
    public static partial CreatorResponseTo ToResponse(this Creator creator);
    public static partial IEnumerable<CreatorResponseTo> ToResponse(this IEnumerable<Creator> creators);
}